napari/_qt/qt_resources/__init__.py

```diff
@@ -0,0 +1,165 @@
+import sys
+from functools import lru_cache
+from glob import glob
+from importlib.util import module_from_spec, spec_from_file_location
+from os import environ, fspath
+from os.path import abspath, dirname, expanduser, join
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Callable, List, Optional, Tuple
+
+from qtpy import API, QT_VERSION
+
+from napari import __version__
+
+from .build_icons import build_pyqt_resources
+
+
+def _try_touch_file(target) -> Optional[Path]:
+    """Test to see if we have permissions to create a file at ``target``.
+
+    If the target already exists, it will not be touched.  If it does not
+    exist, this function attempts to create it and delete it (i.e. testing
+    permissions).  NOTE: all parent directories required to write the file will
+    be created, but NOT deleted.
+
+    If successful, the path is returned, if not, return None.
+
+    Parameters
+    ----------
+    target : str
+        Filepath to test
+
+    Returns
+    -------
+    target : str or None
+        Returns the target if it is writeable, returns None if it is not.
+    """
+    target = Path(target)
+    if not target.exists():
+        try:
+            # create parent directories
+            target.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Won't the parent dirs be left lying around by this function? That seems suboptimal...

Reply:
`target.unlink()` below is supposed to clean it up (though I see that depending on the nature of some exception it might not).  But that code is many months old (just moved from `napari.resources.__init__` and I haven't seen anything lying around yet